cloudbridge/cloud/providers/openstack/provider.py

```diff
@@ -236,19 +238,58 @@ def _connect_cinder(self):
 #         return glance_client.Client(version=api_version,
 #                                     session=self.keystone.session)
 
-    def _connect_swift(self):
+    @staticmethod
+    def _clean_options(options, method_to_match):
+        """
+        Returns a copy of the source options with all keys that are not in the
+        ``method_to_match`` parameter list removed.
+
+        :param options: The source options.
+        :type options: ``dict``
+        :param method_to_match: The method whose signature is to be matched
+        :type method_to_match: A callable
+        :return: A copy of the source options with all keys that are not in the
+            ``method_to_match`` parameter list removed. If options is ``None``
+            then this will be an empty dictionary
+        :rtype: ``dict``
+        """
+        result = dict(options or {})
+        if len(result):
+            # Don't allow the options to override our authentication
+            result['os_options'] = None
+            passed_in_options = set(result.keys())
+            try:
+                method_signature = inspect.signature(method_to_match)
+                parameters = set(method_signature.parameters.keys())
+            except AttributeError:
+                parameters = set(inspect.getargspec(method_to_match)[0])
+            difference = passed_in_options - parameters
+            for name in difference:
+                del result[name]
+        return result
```

Comment:
@MartinPaulo A dictionary comprehension based implementation seems to shorten the code slightly. Do you foresee any issues?
```
def _clean_options(options, method_to_match):
    if options:
        try:
              method_signature = inspect.signature(method_to_match)
              parameters = set(method_signature.parameters.keys())
        except AttributeError:
              parameters = set(inspect.getargspec(method_to_match).args)
        result = { key : val for key, val in options.items() if key not in parameters }
        # Don't allow the options to override our authentication
        result['os_options'] = None
        return result
     else:
         return options or {}
```